### PR TITLE
refactor: refactor the unit tests

### PR DESCRIPTION
**Pull request by vvv**
Pattern 1
---------

Before:
```rust
        assert_eq!(a, true);
        assert_eq!(b, false);
```

After:
```rust
        assert!(a);
        assert!(!b);
```

Rationale: if `assert_eq!(pred, true)` is clearer than `assert!(pred)`,
then `assert_eq!(pred == true, true)` is even more clearer.

Pattern 2
---------

Before:
```rust
    match pred {
        true => {
            // ...
        }
        false => {
            // ...
        }
    }
```

After:
```rust
    if pred {
        // ...
    } else {
        // ...
    }
```

Pattern 3
---------

Before:
```rust
    #[test]
    fn example() {
        // `f` returns `Result<(), Error>`
        assert!(f().is_ok());
    }
```

After:
```rust
    #[test]
    fn example() {
        // `f` returns `Result<(), Error>`
        f().unwrap();
    }
```

Wat
---

Before:
```rust
    #[test]
    fn test_is_router() {
        let is_router = is_router();
        assert_eq!(is_router, false);
    }
```

After:
```rust
    #[test]
    fn test_is_router() {
        assert!(!is_router();
    }
```

### CODE REVIEW

**Comment by vvv** [comment]
`test_index_ok` is still failing (#270).

```
failures:

---- handler::http::router::ui::tests::test_index_ok stdout ----
thread 'handler::http::router::ui::tests::test_index_ok' panicked at 'assertion failed: resp.status().is_success()', src/handler/http/router/ui.rs:70:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    handler::http::router::ui::tests::test_index_ok

test result: FAILED. 114 passed; 1 failed; 2 ignored; 0 measured; 0 filtered out; finished in 0.85s
```

cc @hengfeiyang

**Comment by vvv** [comment]
cc @oasisk

File: src/common/file.rs [hunk 1]
```diff
@@ -38,19 +38,12 @@ pub fn put_file_contents(file: &str, contents: &[u8]) -> Result<(), std::io::Err
     Ok(())
 }
 
-#[inline(always)]
-pub fn delete_file(file: &str) -> Result<(), std::io::Error> {
-    std::fs::remove_file(file)?;
-    Ok(())
-}
-
```
Comment (hengfeiyang) [inline src/common/file.rs:46]: Why did you delete this function?

File: src/common/file.rs [hunk 2]
```diff
@@ -61,19 +54,11 @@ mod tests {
...
-        let resp = scan_files(file_name);
-        assert!(resp.len() > 0);
 
-        let resp = delete_file(file_name);
-        assert!(resp.is_ok());
+        put_file_contents(file_name, content).unwrap();
+        assert_eq!(get_file_contents(file_name).unwrap(), content);
+        assert!(get_file_meta(file_name).unwrap().is_file());
+        assert!(!scan_files(file_name).is_empty());
+        std::fs::remove_file(file_name).unwrap();
```
Comment (hengfeiyang) [inline src/common/file.rs:62]: And test an std library.

File: src/infra/cache/tmpfs.rs [hunk 3]
```diff
@@ -165,16 +165,16 @@ mod tests {
...
     fn create_read_directory() {
-        assert_eq!(true, create_dir_all("/tmp/test_dir/abc/").is_ok());
-        assert_eq!(true, create_dir("/tmp/test_dir/abc/cde/").is_ok());
-        assert_eq!(true, remove_dir("/tmp/test_dir/abc/cde/").is_ok());
-        assert_eq!(true, remove_dir_all("/tmp/test_dir/").is_ok());
+        assert!(create_dir("/tmp/test_dir/abc/cde/").is_err());
+        create_dir_all("/tmp/test_dir/abc/").unwrap();
+        create_dir("/tmp/test_dir/abc/cde/").unwrap();
+        remove_dir("/tmp/test_dir/abc/cde/").unwrap();
+        remove_dir_all("/tmp/test_dir/").unwrap();
```
Comment (hengfeiyang) [inline src/infra/cache/tmpfs.rs:178]: you removed assert maybe cause panic if the function not work.

File: src/infra/storage/local.rs [hunk 4]
```diff
@@ -83,16 +83,12 @@ mod test_util {
         let file_text = "Some text";
         let file_name = "a/b/c/new_file.parquet";
 
-        let resp = local.put(file_name, bytes::Bytes::from(file_text)).await;
-        assert!(resp.is_ok());
-
-        let resp = local.get(file_name).await;
-        assert_eq!(resp.unwrap(), bytes::Bytes::from(file_text));
+        local.put(file_name, bytes::Bytes::from(file_text)).await.unwrap();
```
Comment (hengfeiyang) [inline src/infra/storage/local.rs:86]: We should assert the result, maybe put got error?

File: src/infra/storage/local.rs [hunk 5]
```diff
@@ -83,16 +83,12 @@ mod test_util {
...
+        assert_eq!(local.get(file_name).await.unwrap(), bytes::Bytes::from(file_text));
 
         let resp = local.list("").await;
         assert!(resp.unwrap().contains(&file_name.to_string()));
 
-        let resp = local.del(&[file_name]).await;
-        assert!(resp.is_ok());
+        local.del(&[file_name]).await.unwrap();
```
Comment (hengfeiyang) [inline src/infra/storage/local.rs:92]: same, we should assert the result. unwrap maybe cause panic.

File: src/meta/sql.rs [hunk 6]
```diff
@@ -877,17 +877,13 @@ mod tests {
...
-            }
-            assert!(meta.is_ok());
-            let time_range = meta.unwrap().time_range.unwrap();
-            assert_eq!(time_range.0, sql.1 .0);
-            if sql.1 .1 != 0 {
-                assert_eq!(time_range.1, sql.1 .1);
+            (r#"select * from tbl where time_range("_timestamp", '2022-10-19T15:19:24.587Z','2022-10-19T15:34:24.587Z')"#,(1666192764587000,1666193664587000))].to_vec();
+
+        for (sql, (expected_t1, expected_t2)) in samples {
+            let (actual_t1, actual_t2) = Sql::new(sql).unwrap().time_range.unwrap();
```
Comment (hengfeiyang) [inline src/meta/sql.rs:883]: Same, it's shortly, but we can't directly use unwrap the SQL maybe error.

**Review by hengfeiyang — CHANGES_REQUESTED** [review]

File: src/common/file.rs (hunk 1 above)
Comment (vvv) [reply src/common/file.rs:46]: AFAICS, this function is only ever used in this file's unit test. Functionality-wise, this is a wrapper over stdlib's `std::fs::remove_file`, it doesn't add anything new. Why would we want to create a level of indirection?

Am I missing some context?

Comment (hengfeiyang) [reply src/common/file.rs:46]: Yep, it's just a wrap for std lib and marked as `#[inline(always)] it will replace to std lib when compile. just want to provide a interface that you can create file, write file, delete file by common library.

File: src/common/file.rs (hunk 2 above)
Comment (vvv) [reply src/common/file.rs:62]: Why would we want to test standard library calls? :)

If they did't work, broken unit test would the least of our problems. If we suspected that Rust's standard library is insufficiently tested, we should have switched to different programming language immediately.

File: src/common/file.rs (hunk 1 above)
Comment (hengfeiyang) [reply src/common/file.rs:46]: of cause, we don't use it anymore, we can delete it.

File: src/common/file.rs (hunk 2 above)
Comment (hengfeiyang) [reply src/common/file.rs:62]: i mean why call a std lib here, not let you add a test for std lib.

File: src/infra/cache/tmpfs.rs (hunk 3 above)
Comment (vvv) [reply src/infra/cache/tmpfs.rs:178]: [`Result::unwrap`](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap) does panic on `Err` value.

We shouldn't use `unwrap` in production code or documentation, but unwrapping in tests is totally acceptable.

```rust
pub fn ensure_oddity(n: u8) -> Result<(), String> {
    if n % 2 == 0 {
        Err(format!("{n} is even"))
    } else {
        Ok(())
    }
}

#[test]
fn test_odd() {
    assert!(ensure_oddity(13).is_ok());
    ensure_oddity(7).unwrap();
}

#[test]
#[should_panic]
fn test_even() {
    ensure_oddity(8).unwrap();
}

#[test]
#[should_panic]
fn test_zero() {
    assert!(ensure_oddity(0).is_ok());
}
```

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ba0b934b1b5bfaa40a7654af6d453b43)

Comment (hengfeiyang) [reply src/infra/cache/tmpfs.rs:178]: You added the unwrap.

File: src/infra/storage/local.rs (hunk 4 above)
Comment (vvv) [reply src/infra/storage/local.rs:86]: `res.unwrap()` is semantically equivalent to `assert!(res.is_ok())` — both expressions panic if `res.is_err()`.

```rust
pub const OK: Result<(), ()> = Ok(());
pub const ERR: Result<(), ()> = Err(());

#[test]
fn test_ok() {
    assert!(OK.is_ok());
    OK.unwrap();
}

#[test]
fn test_err1() {
    assert!(ERR.is_err());
}

#[test]
#[should_panic]
fn test_err2() {
    assert!(ERR.is_ok());
}

#[test]
#[should_panic]
fn test_err3() {
    ERR.unwrap();
}
```

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=825cee8a6b7c6ca5a6a2204f6205ef55)

File: src/infra/storage/local.rs (hunk 5 above)
Comment (vvv) [reply src/infra/storage/local.rs:92]: Explained above.

File: src/meta/sql.rs (hunk 6 above)
Comment (vvv) [reply src/meta/sql.rs:883]: If `Sql::new` returns `Err`, we want the test to panic. This code is equivalent to the original, more roundabout, implementation.

https://github.com/zinclabs/zincobserve/blob/14ea6001d6f4229dd49fb53a8affc18962133241/src/meta/sql.rs#L882-L886

BTW, if we want to see `sql` in case of error, we should replace `Sql::new(sql).unwrap()` with `Sql::new(sql).expect("sql={sql}")` or something like this.

**Review by hengfeiyang — APPROVED** [review]
